tests/integration.rs

```diff
@@ -1,8 +1,29 @@
-use std::io::{Read, Write};
+use std::env;
+use std::io::Write;
+use std::path::PathBuf;
 use std::process::{Command, Stdio};
+use tempfile::TempDir;
+
+fn canonicalize_path(path: &str) -> String {
+    PathBuf::from(path)
+        .canonicalize()
+        .unwrap()
+        .to_str()
+        .unwrap()
+        .to_string()
+}
 
 fn run_shell_command(input: &str) -> (String, String) {
-    let mut child = Command::new("./target/debug/aish")
+    let temp_dir = TempDir::new().expect("Failed to create temp dir");
+
+    // Determine the path to the aish executable
+    let mut aish_path = env::current_exe().expect("Failed to get current exe path");
+    aish_path.pop(); // Remove the test executable name
+    aish_path.pop(); // Remove 'deps'
+    aish_path.push("aish"); // Add the aish executable name
```

Comment:
Why?

Reply:
We had to introduce the `temp_dir` to isolate file operations used during tests but this changed the working directory for the test environment. The original command `./target/debug/aish` is a relative path but this path isn't correct now as we are running from the temporary directory. Using the absolute path allows us to find and run the aish executable from the temporary directory.

Reply:
Can we not canonicalize the relative path instead of doing the popping?

Reply:
Found a simpler way of doing it.